### ISSUE TITLE
[Feature] Flip feature flags

### ISSUE THREAD

**Issue by DmitriGekhtman** [ray-project/kuberay#357]
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/kuberay/issues) and found no similar feature requirement.

### Description

With the KubeRay 0.3.0 release coming up, I think it's time to flip the feature flags introduced in
https://github.com/ray-project/kuberay/pull/208 (fix scale-down logic when autoscaling)
cc @sriram-anyscale @me
Flipping this flag is important to simplify configuration for individuals using the autoscaler.
The [test of autoscaling in the Ray CI ](https://github.com/ray-project/ray/blob/1055eadde0b92921fdb42a5e39c7c1ba6c2faadb/python/ray/tests/kuberay/test_autoscaling_e2e.py)I think is sufficient validation that the flag can be flipped.

https://github.com/ray-project/kuberay/pull/231 (enable updating pod configuration)
If additional testing is needed first, we should add that.
cc @wilsonwang371 @je

### Use case

Simplify configuration.

### Related issues

https://github.com/ray-project/kuberay/issues/349

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

**Comment by Jeffwan** [comment]
Agree. For the #208, we can definitely flip the feature flags. For #231, we don't have enough feedbacks from the community yet.

**Comment by DmitriGekhtman** [comment]
This sounds reasonable.

> For https://github.com/ray-project/kuberay/pull/231, we don't have enough feedbacks from the community yet.

^ Let's think about how to gather feedback and test this for the following release.

**Comment by DmitriGekhtman** [comment]
I'll open a separate issue for the update flag -- not critical for the 0.3.0 release, though important to document the current situation.
